synapse/storage/databases/main/account_data.py

```diff
@@ -536,9 +536,9 @@ def _add_account_data_for_user(
         self.db_pool.simple_insert_many_txn(
             txn,
             table="ignored_users",
+            keys=("ignorer_user_id", "ignored_user_id"),
             values=[
-                {"ignorer_user_id": user_id, "ignored_user_id": u}
-                for u in currently_ignored_users - previously_ignored_users
+                (user_id, u) for u in currently_ignored_users - previously_ignored_users
```

Comment:
There's a handful (maybe 1/3rd?) of callers which could benefit from passing in some constant values for some keys, but I figured that could be left to another round of optimization, if we wanted to complicate the API. (I'm unsure if there's real benefit to it...)